### PR TITLE
1290 update

### LINKED ISSUE #227: Version support for CPLEX 12.9
According to dep/builds.jl, the current version of this module does not support CPLEX 12.9

### PR DESCRIPTION
**Pull request by jd-lara**
Update `CPLEX.jl` for version 12.9. The changes are related to the changes in the log API see [12.9 Changes](https://www.ibm.com/support/knowledgecenter/en/SSSA5P_12.9.0/ilog.odms.studio.help/CPLEX/ReleaseNotes/topics/releasenotes1290/removed.html). 

The MOI tests for linear10b are not passing, from reading the testing code I inferred that the affine interval tests were not fully implemented https://github.com/JuliaOpt/CPLEX.jl/blob/bb3f45c236fb0afb2fa2ed75c448e17b9a64a61e/test/MathOptInterface/MOI_wrapper.jl#L12 

Otherwise, all the tests pass except for linear10b

Closes #227

### CODE REVIEW

**Comment by odow** [comment]
You should add 10b to the exclude list, or wrap it in a `MOI.Bridges.IntervalBridge` layer.

**Comment by odow** [comment]
It looks like there is one breaking change to setting the log file:

https://github.com/JuliaOpt/CPLEX.jl/blob/bb3f45c236fb0afb2fa2ed75c448e17b9a64a61e/src/cpx_env.jl#L53-L63

It's now `setlogfilename`, and `CPXfopen` has been removed.

https://www.ibm.com/support/knowledgecenter/en/SSSA5P_12.9.0/ilog.odms.studio.help/CPLEX/ReleaseNotes/topics/releasenotes1290/removed.html

File: src/cpx_env.jl [hunk 1]
```diff
@@ -50,6 +50,9 @@ function close_CPLEX(env::Env)
     end
 end
 
+#= CPLEX API's used in this function are not longer supported and was deprecated since 12.8
```
Comment (jd-lara) [inline src/cpx_env.jl]: @odow this is the code to fix the breaking change.

Comment (odow) [reply src/cpx_env.jl]: Oops, I scrolled right past this. But this will now break earlier versions? What minimum version of CPLEX should we support?

Comment (jd-lara) [reply src/cpx_env.jl]: This change will make the minimum version supported 12.8 

https://www.ibm.com/support/knowledgecenter/en/SSSA5P_12.8.0/ilog.odms.studio.help/CPLEX/ReleaseNotes/topics/releasenotes128/deprecated.html

There is no mention of this change in the release notes for 12.71 
https://www.ibm.com/support/knowledgecenter/en/SSSA5P_12.7.1/ilog.odms.studio.help/CPLEX/ReleaseNotes/topics/releasenotes1271/deprecated.html

**Comment by jd-lara** [comment]
@odow the test for `linear10b` have been wrapped in an MOIB. All tests are passing. 

image

File: src/cpx_env.jl (hunk 1 above)
Comment (odow) [reply src/cpx_env.jl]: If so, we should update
https://github.com/JuliaOpt/CPLEX.jl/blob/bb3f45c236fb0afb2fa2ed75c448e17b9a64a61e/deps/build.jl#L26

Comment (jd-lara) [reply src/cpx_env.jl]: Actually, we should move the current master to a branch `pre-1280` and drop the support for versions older than 12.80 in master.

**Comment by jd-lara** [comment]
The build function still works well 
image

File: src/cpx_env.jl (hunk 1 above)
Comment (odow) [reply src/cpx_env.jl]: I think it's reasonable to remove pre 12.8 support. If people complain, they can volunteer to maintain the code.

File: src/cpx_env.jl [hunk 2]
```diff
@@ -61,6 +64,15 @@ function set_logfile(env::Env, filename::String)
     throw(CplexError(env, stat))
   end
 end
+=#
```
Comment (odow) [inline src/cpx_env.jl]: This comment can be removed now that we're deprecating.

File: test/C_API/low_level_api.jl [hunk 3]
```diff
@@ -2,4 +2,4 @@
     env = CPLEX.Env()
     CPLEX.set_logfile(env, "cplex.log")
     CPLEX.close_CPLEX(env)
-end
+end
```
Comment (odow) [inline test/C_API/low_level_api.jl]: Super nit: end files with a new line

File: src/cpx_env.jl (hunk 1 above)
Comment (jd-lara) [reply src/cpx_env.jl]: Ok, will also update the README file

**Comment by odow** [comment]
Let me know when you're good to merge.

**Comment by jd-lara** [comment]
If you are ok with the README additions, then I am ready.

File: README.md [hunk 4]
```diff
@@ -49,7 +49,7 @@ julia> Pkg.build("CPLEX")
 Help! I got `LoadError: Unable to locate CPLEX installation`
 ----------------------------------
 
-Which version of CPLEX are you trying to install? Currently CPLEX.jl only supports 1260, 1261, 1262, 1263, 1270, 1271, and 1280. If it's not one of those, [file an issue](https://github.com/JuliaOpt/CPLEX.jl/issues/new) with the version number you'd like to support. Some steps need to be taken (like checking for new or renamed parameters) before CPLEX.jl can support new versions.
+Which version of CPLEX are you trying to install? Currently, CPLEX.jl only supports 1280 and 1290 given recent changes [the API](https://www.ibm.com/support/knowledgecenter/en/SSSA5P_12.9.0/ilog.odms.studio.help/CPLEX/ReleaseNotes/topics/releasenotes1290/removed.html). If your CPLEX version is < 1280, CPLEX.jl compatibility with older versions has been kept in the branch `pre-1280` but won't be maintained. If you want to support newer versions of CPLEX not listed above, [file an issue](https://github.com/JuliaOpt/CPLEX.jl/issues/new) with the version number you'd like to support. Some steps need to be taken (like checking for new or renamed parameters) before CPLEX.jl can support new versions.
```
Comment (odow) [inline README.md]: > If your CPLEX version is < 1280, CPLEX.jl compatibility with older versions has been kept in the branch `pre-1280` but won't be maintained.

How about:

If your CPLEX version is < 1280, you can add the last supported version of CPLEX.jl via `julia> ] add CPLEX@0.4`. However, we recommend that you upgrade your version of CPLEX.

Comment (jd-lara) [reply README.md]: LGTM

**Pull request merged**